tests/RobotFramework/tests/tedge_to_te_converter/convert_tedge_topics_to_te_topics.robot

```diff
@@ -18,32 +18,29 @@ Convert main device measurement topic
 
 Convert child device measurement topic
     Execute Command    tedge mqtt pub tedge/measurements/child '{"temperature":25}'
-    Should Have MQTT Messages    te/device/child///m/    message_pattern={"temperature":25}   
+    Should Have MQTT Messages    te/device/child///m/    message_pattern={"temperature":25}
 
 Convert main device event topic
     Execute Command    tedge mqtt pub tedge/events/login_event '{"text":"someone logedin"}'
     Should Have MQTT Messages    te/device/main///e/login_event    message_pattern={"text":"someone logedin"}
 
 Convert child device event topic
-    Execute Command    tedge mqtt pub tedge/events/child/login_event '{"text":"someone logedin"}'
+    Execute Command    tedge mqtt pub tedge/events/login_event/child '{"text":"someone logedin"}'
     Should Have MQTT Messages    te/device/child///e/login_event    message_pattern={"text":"someone logedin"}
 
 Convert main device alarm topic
-    Execute Command    tedge mqtt pub tedge/alarms/minor/test_alarm '{"text":"test alarm"}'
-    Should Have MQTT Messages    te/device/main///a/test_alarm    message_pattern={"text":"test alarm","severity":"minor"}    
+    Execute Command    tedge mqtt pub tedge/alarms/minor/test_alarm '{"text":"test alarm"}' -q 2 -r
+    Should Have MQTT Messages    te/device/main///a/test_alarm    message_contains=severity  message_contains=minor
```

Comment:
@reubenmiller Do we have a way to compare json values? and not strings?

Reply:
> No, but I'm happy to add something in that can accept a dictionary, and it will do a deep object comparison.

That would be super helpful for a scenario like this.

> 
> Btw, the suggested change won't work because `message_contains` only accepts a string (not an array of strings), below is the function signature.
> 
> ```python
> @keyword("Should Have MQTT Messages")
>     def mqtt_should_have_topic(
>         self,
>         topic: str,
>         date_from: relativetime_ = None,
>         date_to: relativetime_ = None,
>         message_pattern: str = None,
>         message_contains: str = None,
>         minimum: int = 1,
>         maximum: int = None,
>         **kwargs,
>     ) -> List[Dict[str, Any]]:
> ```

But, I can just check one string right?, say for example just `severity`, IMO that would be sufficient for now, if implementing a new API takes longer.

Reply:
> I'm happy to add something in that can accept a dictionary, and it will do a deep object comparison.

This can be done later. This fix is rather urgent.

> Btw, the suggested change won't work because message_contains only accepts a string (not an array of strings)

You mean that one can check either that `message_contains=severity` or `message_contains=minor` but not both?

Reply:
This is where the doc strings are helpful, yes, `message_contains=` does not support multiple values, however you can `message_pattern=` which accepts a regular expression (which you can then make to accept multiple values).

For example:

```
message_pattern="value":\s*\d+
```

Reply:
We need to move fast - as the system tests are broken on man.

=> I simplified the tests for alarms, only checking that the severity is inserted in the message payload.